applications/FluidDynamicsApplication/custom_processes/apply_mach_dependent_boundary_conditions.h

```diff
@@ -0,0 +1,314 @@
+//
+//  Main author:     Eduard Gómez
+//
+
+#if !defined(KRATOS_APPLY_MACH_DEPENDENT_BOUNDARY_CONDITIONS_H_INCLUDED )
+#define  KRATOS_APPLY_MACH_DEPENDENT_BOUNDARY_CONDITIONS_H_INCLUDED
+
+
+
+// System includes
+#include <string>
+#include <iostream>
+
+
+// External includes
+
+
+// Project includes
+#include "includes/define.h"
+#include "includes/kratos_parameters.h"
+#include "includes/model_part.h"
+#include "processes/process.h"
+#include "containers/model.h"
+#include "utilities/interval_utility.h"
+namespace Kratos
+{
+  ///@addtogroup FluidDynamicsApplication
+  ///@{
+
+  ///@name Kratos Globals
+  ///@{
+
+  ///@}
+  ///@name Type Definitions
+  ///@{
+
+  ///@}
+  ///@name  Enum's
+  ///@{
+
+  ///@}
+  ///@name  Functions
+  ///@{
+
+  ///@}
+  ///@name Kratos Classes
+  ///@{
+
+  /** 
+   * @brief This process applies diferent boundary conditions accoring to the
+   * mach regime.
+   * Parameters:
+   * @param model_part_name The model part onto which to enforce the boundary
+   *    conditions.
+   * @param flow_direction_variable A variable to use to know direction of the
+   *    flow. Its magnitude will me ignored. must be historical.
+   * @param refresh_normals_every_time_step Whether to recompute the normal
+   *    direction every time-step.Leave at false unless your mesh or geometry
+   *    is expected to change.
+   * @param subsonic_boundary_conditions A list of variables to fix when the
+   *    mach projected onto the normal of the boundary is below 1.
+   * @param supersonic_boundary_conditions  A list of variables to fix when
+   *    the mach projected onto the normal of the boundary is above 1.
+   */
+  class KRATOS_API(FLUID_DYNAMICS_APPLICATION) ApplyMachDependentBoundaryConditions: public Process
+  {
+  public:
+        ///@name Type Definitions
+        ///@{
+        //
+        typedef ModelPart::NodeType NodeType;
+        typedef Variable<array_1d<double, 3>> VectorVariable;
+        
+        /**
+         * @brief This class validates and manages a variable to fix
+         * and the value to fix it to.
+         * 
+         * Method ActivateIfInsideTimeInterval must be called at the beginning
+         * of the time step to see if the BC is active or not. This will cause
+         * the following behaviour in the other two public methods:
+         * - Enforce:
+         *     - If t inside interval  -> Fix dof
+         *     - If t outside interval -> Do nothing
+         * - Free
+         *     - Always -> Free dof
+         * 
+         * This allows for checking the time only once at the beginning of the
+         * time-step rather than at every node.
+         */
+        class BoundaryConditionUtility
+        {
+        public:
+            BoundaryConditionUtility(const std::string & variable_name,
+                                     const double Value,
+                                     const IntervalUtility & rIntervalUtility);
```

Comment:
```suggestion
            BoundaryConditionUtility(
                const std::string& rVariableName,
                const double Value,
                const IntervalUtility & rIntervalUtility);
```
Please follow the style guide convention (see [here](https://github.com/KratosMultiphysics/Kratos/wiki/Style-Guide))